satip/goes_download_manager.py

```diff
@@ -0,0 +1,96 @@
+"""
+Script for downloading GOES data.
+"""
+
+import datetime
+import logging
+import os
+
+from goes2go import GOES
+
+
+class GOESDownloadManager:
+    """
+    Manager class for downloading GOES data.
+    """
+    def __init__(self, data_dir, log_directory=None):
+        """
+        Initialize the GOESDownloadManager.
+
+        Args:
+            data_dir (str): Directory to save downloaded GOES data.
+            log_directory (str, optional): Directory to save logs.
+            If None, logging is printed to STDOUT.
+        """
+        self.data_dir = data_dir
+        self.ensure_directory_exists(self.data_dir)
+
+        if log_directory:
+            self.ensure_directory_exists(log_directory)
+            logging.basicConfig(
+                filename=os.path.join(log_directory, 'goes_download.log'),
+                level=logging.INFO)
+        else:
+            logging.basicConfig(level=logging.INFO)
+
+        logging.info(f"GOESDownloadManager initialized. Data will be saved to: {data_dir}")
+
+    @staticmethod
+    def ensure_directory_exists(directory):
+        """Ensures the specified directory exists, creating it if necessary."""
+        if not os.path.exists(directory):
+            try:
+                os.makedirs(directory)
+                logging.info(f"Created directory: {directory}")
+            except Exception as e:
+                logging.error(f"Error creating directory {directory}: {e}")
+                raise
+    def download_goes_data(self, start_time, end_time, product='ABI-L1b-RadC',
```

Comment:
It seems the product is simply named `ABI-L1b-Rad`, and the domain could be either `F`, `C`, or `M`. By default, the product should be labeled as `ABI-L1b-Rad`. It might be beneficial to revisit these details to confirm they align correctly.